lib/internal/modules/esm/loader.js

```diff
@@ -45,6 +49,9 @@ const {
   initializeImportMeta
 } = require('internal/modules/esm/initialize_import_meta');
 const { defaultLoad } = require('internal/modules/esm/load');
+const {
+  defaultGetFileSystem
+} = require('internal/modules/esm/get_file_system');
```

Comment:
Is the plan to only use this in the `esm/loader.js` and not in the CJS loader? That seems like it might limit the applicability of these hooks, especially when CJS interfaces with eg exports resolutions.

Reply:
At the moment the other hooks are only used in the esm code path so I didn't want to change that since it's not required strictly speaking. A followup PR however will be to consider `getFileSystem` for cjs.